projects/storefrontlib/cms-components/content/video/video.component.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  AfterViewChecked,
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  ElementRef,
+  HostBinding,
+  OnDestroy,
+  Renderer2,
+  ViewChild,
+} from '@angular/core';
+import {
+  CmsBannerComponentMedia,
+  CmsResponsiveBannerComponentMedia,
+  CmsService,
+  CmsVideoComponent,
+  ContainerBackgroundOptions,
+  ContainerSizeOptions,
+  PageType,
+  SemanticPathService,
+} from '@spartacus/core';
+import { Observable, Subscription } from 'rxjs';
+import { take, tap } from 'rxjs/operators';
+import { CmsComponentData } from '../../../cms-structure/page/model/cms-component-data';
+import {
+  Media,
+  MediaContainer,
+} from '../../../shared/components/media/media.model';
+import { MediaService } from '../../../shared/components/media/media.service';
+
+@Component({
+  selector: 'cx-video',
+  templateUrl: './video.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VideoComponent implements AfterViewChecked, OnDestroy {
+  @HostBinding('class') styleClasses: string | undefined;
+  @ViewChild('videoPlayer') videoPlayer: ElementRef;
+
+  source: string | undefined;
+  thumbnail: Media | undefined;
+  routerLink: string | any[] | undefined;
+  autoPlay: boolean;
+  loop: boolean;
+  mute: string | undefined;
+  height: string;
+  protected subscriptions = new Subscription();
+
+  data$: Observable<CmsVideoComponent> = this.component.data$.pipe(
+    tap((data) => {
+      this.styleClasses = data.styleClasses;
+      this.setMedia(data.video, data.videoMedia, data.containerBackground);
```

Comment:
can we just pass `data` to these functions?

Reply:
I was debating on this too, changed